docs/modeling-your-data/modeling-your-data-with-dbt/dbt-event-splitting-model/index.md

```diff
@@ -0,0 +1,111 @@
+  - `event_2` _(option - object)_
+  - ...
+  - `event_n` _(option - object)_
+- `users` _(option - array)_: Array of strings of schemas for your user contexts to add to your users table as columns, if not provided will not generate users model
+
+An example configuration can be found in the `utils/example_event_split_config.json` file.
+
+### Producing your models
+To produce your models you need to run `python utils/snowplow_split_events_model_gen.py path/to/your/config` from the root of your dbt project. This will produce one `.sql` file for each of the event names specified in the `events` part of your configuration, one `.sql` file for the combined filtered events table if a name was provided, and one `.sql` file for your users table if schema(s) were provided. These files will be in your `models` folder in the sub-folder specified in your config. 
+
+:::info
+Custom error messages have been added to the script to try and catch any issues and provide suggested resolutions to any issues such as invalid configurations or failing validation of schemas. If you persist in getting errors when validating schemas you believe you be correct, you can disable this validation by setting `validate_schemas` to `false` in your config. 
+:::
+
+### Adding new models
+To add new models for new types of events, simply add them to your config and run the model again. Note that `overwrite` must be set to `true` for your filtered events table model to include these new events, and that this will overwrite any manual changes you made to the other models generated by this script.
+
+### Updating your models
+If you wish to update your models, such as adding a new context or removing a column, you must update your config file and run the script again, ensuring the `overwrite` value is set to `true`. Note that this will remove any custom changes you made to the model file itself and you will need to re-add these.
+
+All events will be updated including the filtered events table, it is not possible at this time to update just a subset of models.
+
+:::warning
+Adding or removing columns from your dbt model without specifying the `on_schema_change` model configuration, or directly altering the tables in your database, is very likely to result in unintended outcomes - either your new columns will not appear or dbt may return an error. Please see the [dbt docs](https://docs.getdbt.com/docs/build/incremental-models#what-if-the-columns-of-my-incremental-model-change) for what to do in this situation. If you need to backfill new columns regardless you can perform a rerun of the entire model following the instructions [here](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-custom-models/index.md#tearing-down-and-restarting-a-subset-of-models) *(note using the `--full-refresh` flag will not work in this case due to the use of the Snowplow incremental logic)*.
+:::
+
+### Removing models
```

Comment:
Worth adding that you can just drop them in SQL or no?

Reply:
They can't just drop the tables to do a refresh right? Because it would still be in the manifest table?